test/util/autodiff/numeric_derivative.hpp

```diff
@@ -34,7 +34,7 @@ namespace ngraph
         /// \returns vector of dy/dvar, where each dy/dvar's shape is concat(y.shape(), var.shape())
         template <typename T>
         std::vector<std::shared_ptr<runtime::Tensor>>
-            numeric_derivative(const std::shared_ptr<runtime::Backend>& backend,
+            numeric_derivative(runtime::Backend& backend,
```

Comment:
@rkimballn1 Is it necessary to eliminate `const` here?

Reply:
`const shared_ptr<X>` is a `const shared_ptr`, not a `const X`. I did not get rid of const on X as it was never there. I did try making this const but the functions compile() and call() on backend are not const so backend can't be const.

Reply:
@earhart I like your rule. Right now we pretty much pass everything around everywhere as shared_ptr which makes me crazy. I will talk this idea over with a few folks here to see if we can come to a conclusion.

Reply:
@earhart I talked with Scott and we are both meh about the pointer vs reference. I *did* change it to pointer because it was a pointer before and it reduces the changes on other lines of code.